src/osp/Active/ActiveScene.h

```diff
@@ -332,4 +344,47 @@ struct ACompCamera
 };
 
 
+template<typename FUNC_T>
+void ActiveScene::hierarchy_traverse(ActiveEnt root, FUNC_T callable)
+{
+    using osp::active::ACompHierarchy;
+
+    std::vector<ActiveEnt> parentNextSibling;
```

Comment:
For very small "stacks" like this, it seems to me like we need a generalized stack allocator.

Take a look here

 https://github.com/GameFoundry/bsf/blob/master/Source/Foundation/bsfUtility/Allocators/BsStackAlloc.h
 https://github.com/GameFoundry/bsf/blob/master/Source/Foundation/bsfUtility/Allocators/BsStackAlloc.cpp

Stack allocations are essentially free at runtime, so having a mechanism like that would be very nice.